images/management/commands/import-tandemvault-tags.py

```diff
@@ -0,0 +1,157 @@
+from django.core.management.base import BaseCommand
+from images.models import *
+
+import csv
+import datetime
+import json
+import logging
+import timeit
+
+from progress.bar import Bar
+from unidecode import unidecode
+
+
+class Command(BaseCommand):
+    help = 'Imports all existing image tags from UCF\'s Tandem Vault instance.'
+
+    progress_bar = Bar('Processing')
+    source       = 'Tandem Vault'
+    tags         = []
+    tag_count    = 0
+    tags_created = 0
+    tags_updated = 0
+    tags_deleted = 0
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--file',
+            type=file,
+            help='CSV file containing existing tag and synonym information from Tandem Vault',
+            dest='file',
+            required=True
+        ),
+        parser.add_argument(
+            '--delete-stale',
+            type=bool,
+            help='Whether or not stale tags (tags not assigned to any Images) should be deleted.',
+            dest='delete-stale',
+            default=False,
+            required=False
+        )
+
+    def handle(self, *args, **options):
+        self.tandemvault_tags_csv = options['file']
+        self.do_delete_stale = options['delete-stale']
+
+        # Start a timer for the bulk of the script
+        self.exec_time = 0
+        self.exec_start = timeit.default_timer()
+
+        # Process the CSV
+        self.load_tandemvault_tags()
+        self.process_tandemvault_tags()
+
+        # Delete stale Tandem Vault image tags
+        if self.do_delete_stale:
+            self.delete_stale()
+
+        # Stop timer
+        self.exec_end = timeit.default_timer()
+        self.exec_time = datetime.timedelta(seconds=self.exec_end - self.exec_start)
+
+        # Print the results
+        self.progress_bar.finish()
+        self.print_stats()
+
+        return
+
+    '''
+    Load a CSV of Tandem Vault tags + synonyms.
+    '''
+    def load_tandemvault_tags(self):
+        try:
+            csv_reader = csv.DictReader(self.tandemvault_tags_csv)
+        except csv.Error:
+            logging.error(
```

Comment:
Is there an inner exception or message we can print to give us a clue what the problem with the CSV is?

Reply:
Updated CSV error handling to spit out errors coming from the CSV lib, and added some missing sanity checks to make sure the file looks like a CSV before looping through it.